Kudu.SiteManagement/SiteManager.cs

```diff
@@ -347,6 +345,57 @@ public bool RemoveSiteBinding(string applicationName, string siteBinding, SiteTy
             }
         }
 
+        public async Task StartAppPool(string applicationName)
+        {
+            using (ServerManager iis = GetServerManager())
+            {
+                string appPoolName = GetAppPool(applicationName);
+                var kuduAppPool = iis.ApplicationPools[appPoolName];
+
+                if (kuduAppPool.State != ObjectState.Started)
```

Comment:
if the appPool is in Starting state, it's already in a failure situation.   Whether this appPool start succeeds or fails will not worsen the situation.